account_cash_discount_write_off/models/account_payment_line.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class PaymentLine(models.Model):
+
+    _inherit = 'account.payment.line'
+
+    @api.multi
+    def check_cash_discount_write_off_creation(self):
+        self.ensure_one()
+        move_line = self.move_line_id
+        has_partial_reconciliation = (
+            bool(move_line.matched_debit_ids) or
+            bool(move_line.matched_credit_ids))
+        return (
+            self.pay_with_discount and
+            has_partial_reconciliation and
+            move_line.invoice_id
+        )
+
+    @api.multi
+    def get_cash_discount_writeoff_move_values(self):
+        self.ensure_one()
+        move_line = self.move_line_id
+        partner = move_line.partner_id
+        partner = partner._find_accounting_partner(partner)
+        invoice = move_line.invoice_id
+        company = invoice.company_id
+        tax_adjustment = company.cash_discount_base_amount_type == 'total'
+
+        woff_account = False
+        woff_journal = False
+
+        if invoice:
```

Comment:
why this test?

Reply:
I agree that this method is called only when ```check_cash_discount_write_off_creation``` returns true. And this method returns True if the move line is linked to an invoice. So yes, this test might have no sense. but then I'm sure there won't be a huge stacktrace in the future because of a specific case.